sklearn/feature_selection/from_model.py

```diff
@@ -1,14 +1,78 @@
-# Authors: Gilles Louppe, Mathieu Blondel
+# Authors: Gilles Louppe, Mathieu Blondel, Maheshakya Wijewardena
 # License: BSD 3 clause
 
 import numpy as np
 
-from ..base import TransformerMixin
+from .base import SelectorMixin
+from ..base import (TransformerMixin, BaseEstimator, clone,
+    MetaEstimatorMixin)
 from ..externals import six
-from ..utils import safe_mask, check_array
+
+from ..utils import safe_mask, check_array, deprecated
 from ..utils.validation import NotFittedError, check_is_fitted
 
 
+def _get_feature_importances(estimator, X):
+    """Retrieve or aggregate feature importances from estimator"""
+    if hasattr(estimator, "feature_importances_"):
+        importances = estimator.feature_importances_
+
+    elif hasattr(estimator, "coef_"):
+        if estimator.coef_.ndim == 1:
+            importances = np.abs(estimator.coef_)
+
+        else:
+            importances = np.sum(np.abs(estimator.coef_), axis=0)
+
+    else:
+        raise ValueError("Missing `feature_importances_` or `coef_`"
+                         " attribute, did you forget to set the "
+                         "estimator's parameter to compute it?")
```

Comment:
We should update this docstring as it relates to the old `compute_feature_importances` parameter we had. I would rephrase the error by hinting that the estimator (the provided one or the meta- one) should be fit before calling `transform`.